generate_build_config.py

```diff
@@ -73,11 +86,14 @@ def main():
     parser = argparse.ArgumentParser()
     parser.add_argument('output_filename')
     parser.add_argument('--customisation-script', dest='custom_script')
+    parser.add_argument('--binary-customisation-script',
+                        dest='binary_custom_script')
```

Comment:
In terms of README updates, I'm worried about making that document too long or difficult to parse, as that's the introduction that users will have to the project.

Most users won't care about binary customisations, so I don't want to dilute the clear journey in the README.md.

That said, we do need to think about more detailed documentation, but I'd like to have that outside the scope of this PR.

(I should add a help message though, so let me do that.)